accounting.js

```diff
@@ -0,0 +1,49 @@
+const log4js = require('log4js');
+const logger = log4js.getLogger('<filename>');
+log4js.configure({
+    appenders: {
+        file: { type: 'fileSync', filename: 'logs/debug.log' }
+    },
+    categories: {
+        default: { appenders: ['file'], level: 'debug'}
+    }
+});
+const moment = require('moment');
+
+function doAccounts(dataFile) {
+    let namesList = [];
+    for (let i = 0; i < dataFile.length; i++) {
+        namesList.push(dataFile[i].From);
+        namesList.push(dataFile[i].To);
+    }
+    let accountsList = Array.from(new Set(namesList));
+    
+    let accounts = [];
+    for (let i = 0; i < accountsList.length; i++) {
+        accounts[i] = {Name: accountsList[i], Net_Balance: 0, Transactions: []};
+    }
+
+    for (let i = 0; i < dataFile.length; i++) {
+        for (let accountsnum = 0; accountsnum < accounts.length; accountsnum++) {

```

Comment:
Following on from the comment above, you can save yourself having to do this search by converting `accounts` into a dictionary. Assuming that's done (as above), then this is reduced to:

    dataFile.forEach(dataLine => {
    })